### PR TITLE
Update to latest Theia and GLSP version

### PR DESCRIPTION
**Pull request by ndoschek**
- Update to latest Theia and GLSP dependencies
- Switch to peerDependencies for Theia dependencies and remove resolutions block
- Update missing icons to codicons
- Adapt CSS styling to Theia native styling for comparison dialog and action widget

Part of https://github.com/eclipse-emfcloud/coffee-editor/issues/499

### CODE REVIEW

File: client/comparison-extension/package.json [hunk 1]
```diff
@@ -45,6 +42,12 @@
     "semver": "^6.3.0",
     "typescript": "~4.5.5"
   },
+  "peerDependencies": {
+    "@theia/core": "^1.34.1",
+    "@theia/process": "^1.34.1",
+    "@theia/workspace": "^1.34.1",
+    "@theia/git": "^1.34.1"
+  },
```
Comment (sgraband) [inline client/comparison-extension/package.json]: Are we using any new features that require this version? I am not an expert, but from my understanding, this prevents people from using the application with theia versions lower than the one specified here. So maybe we shouldnt't be too restrictive?

**Review by sgraband — COMMENTED** [review]
Thank you for the changes! Looks good and works great :+1: I just have one inline comment and another general comment:

It seems like for the lower tables we get 2 scrollbars. While this seems to be not a problem with this fix, maybe you have an idea what might cause this?

File: client/comparison-extension/package.json (hunk 1 above)
Comment (ndoschek) [reply client/comparison-extension/package.json]: Thanks, yes that's totally true, I left them at their previous value now (`^1.0.0)

**Comment by ndoschek** [comment]
Thanks @sgraband for your review!
The scrollbars were shown due the fact that the additional header (e.g. `Differences overview`) decreased the container's height. I adapted the height of the tree container and scrollbars should now appear as expected.

**Review by sgraband — APPROVED** [review]
Thank you very much! Works like a charm :+1:

**Pull request merged**